src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/BuildIntegratedProjectAction.cs

```diff
@@ -65,9 +73,14 @@ public class BuildIntegratedProjectAction : NuGetProjectAction
             IReadOnlyList<SourceRepository> sources,
             IReadOnlyList<NuGetProjectAction> originalActions,
             BuildIntegratedInstallationContext installationContext,
-            VersionRange versionRange)
+            VersionRange? versionRange)
             : base(packageIdentity, nuGetProjectActionType, project, sourceRepository: null, versionRange)
         {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
```

Comment:
How about modifying the obsolete constructor to call the new one? This approach would allow us to centralize the null checks in one place.

Reply:
The null checks are different across the constructors. We'd end up with an additional constructor.

Reply:
Can we extract all the common null checks into another method and invoke that method from the constructor?

Here's a suggested implementation of the `ValidateArguments` method:

```cs
private void ValidateArguments(NuGetProject project, PackageIdentity packageIdentity, LockFile originalLockFile, RestoreResultPair restoreResultPair, IReadOnlyList<SourceRepository> sources, object actions, BuildIntegratedInstallationContext? installationContext = null)
{
    if (project == null)
        throw new ArgumentNullException(nameof(project));
    if (originalLockFile == null)
        throw new ArgumentNullException(nameof(originalLockFile));
    if (restoreResultPair == null)
        throw new ArgumentNullException(nameof(restoreResultPair));
    if (sources == null)
        throw new ArgumentNullException(nameof(sources));
}
```

We can add null checks that are specific to a particular constructor within that constructor itself, after validating the common arguments.

We can remove the following null check from the constructors in this class because the [base constructor takes care of that](https://github.com/NuGet/NuGet.Client/blob/d85c6ea4588c5b4f2b4a4d7bf8073837fb308f58/src/NuGet.Core/NuGet.PackageManagement/NuGetProjectAction.cs#L49-L52):

```cs
if (packageIdentity == null)
    throw new ArgumentNullException(nameof(packageIdentity));
```

Reply:
I can clean that up in a future PR.